zenoh-flow/src/runtime/dataflow/loader.rs

```diff
@@ -149,13 +149,13 @@ unsafe fn load_lib_sink(path: PathBuf) -> ZFResult<(Library, Arc<dyn Sink>)> {
 
     Ok((library, (decl.register)()?))
 }
-
-pub fn make_file_path(uri: Url) -> PathBuf {
+fn make_file_path(uri: Url) -> ZFResult<PathBuf> {
     let mut path = PathBuf::new();
-    if let Some(host) = uri.host_str() {
-        path.push(host);
-    }
-    path.push(uri.path());
-
-    path
+    let file_path = match uri.host_str() {
```

Comment:
I really do not like this, but it is needed to let the `PathBuf` understand it is a relative path.
In the case of a relative path, the `host_str` is `.`, so concatenating `host_str` with `path` gives the correct relative path.
Then `std::fs::canonicalize` retrieves the absolute path, and verifies if the file exists.